### DISCUSSION TITLE
Interaction testing issue (with ts-jest)

### DISCUSSION THREAD

**Discussion by fjmillman** [storybookjs/storybook#17590] [Help] [answered]
I'm currently having an issue with running interaction tests for storybook (fantastic work on these libraries by the way!) when using `test-storybook`. The actual interaction test runs successfully in the storybook itself no problem.

Test Failure: https://github.com/fjmillman/expo-nextjs-storybook/runs/5351552510?check_suite_focus=true

When running `test-storybook`, the storybook tests fail with:
- `SyntaxError: Cannot use import statement outside a module` for imports in `WelcomeButton/stories.tsx` and `WelcomeButton/index.tsx`

And gives warnings for:
- `TypeError: Cannot read property 'parent' of undefined` which appears to derive from `@storybook/instrumenter` from it's cjs code

When running `jest` itself which picks up `WelcomeButton/test.tsx` this runs just fine and passes when checking `WelcomeButton/index.tsx`. So I suspect there's an issue around how ts-jest is transpiling code correctly for `jest` but not for `test-storybook`.

Repository: https://github.com/fjmillman/expo-nextjs-storybook
Reproduction: https://621bab1c57a4e5003a1598cb-ptmakjyzib.chromatic.com/?path=/story/components-default--default

It's difficult to tell if this is a bug or just a configuration issue, but I've been struggling to figure out the solution to this issue. I'm happy to open a bug report if this is a bug, but I've left it here in Q&A to see if anyone has any ideas/clues, any help would be much appreciated!

**Comment by shilman** [comment, marked as answer]
It looks like the test runner's jest config is set to run on `*.stories.[jt]sx?`:

https://github.com/storybookjs/test-runner/blob/next/src/config/jest-playwright.ts#L9

When I edit the regex in @fjmillman 's `node_modules` it works.

@yannbf @tmeasday should we relax this to run on `stories.[jt]sx?`? Or make this driven from `main.js`? (@yannbf I thought you already did something like this?)

**Reply by fjmillman** [reply]
Fantastic!

The solution here is to use the `*.stories.[jt]sx?` format in the meantime until the `*stories.[jt]sx?` format is supported in jest-playwright as it is currently in storybook.

Thank you for your help here @shilman

**Reply by yannbf** [reply]
Great finding @shilman and thank you so much for your very thorough issue description @fjmillma! Would you mind explaining what's the reason for defining your stories as `stories.jsx`? Is it to follow the pattern of `MyComponent/(index|style|stories|test).tsx`?

Thing is we have the `*.stories.*` pattern defined as a good practice and our tools support that format. As soon as you don't follow that format you might end up having weird behaviors like the one mentioned in this issue. [Related issue regarding out-of-standard naming](https://github.com/storybookjs/eslint-plugin-storybook/issues/77)

@shilman I worked on respecting main.js globs for the `testMatch` property, and I believe it's not as simple for the `transform` property. There's a bigger question here which is whether we want to support the `stories.*` format, as that would require changes in some tools around the ecosystem like the test-runner, the eslint-plugin and likely others

**Reply by fjmillman** [reply]
@yannbf No problem! We should all be striving to make life much easier for you guys as maintainers, you've all done such an incredible job with Storybook and it's expanding ecosystem and long may that continue.

That is correct, it was to follow the pattern `MyComponent/(index|styles|stories|test).*`. I've since reverted to using the `*.stories.*` format instead given the above issue. I should say, I only moved to using `stories.*` after support was added for it in Storybook, so it was only natural to assume that was available in all cases, but of course this was not the case with regards to Jest testing.
